src/mesh_tools.rs

```diff
@@ -333,6 +334,158 @@ mod tests {
+        v( 1.0,  1.0,  1.0, position, scale), //2
+        v( 1.0,  1.0, -1.0, position, scale), //3
+        // front
+        v(-1.0, -1.0, -1.0, position, scale), //4
+        v( 1.0, -1.0, -1.0, position, scale), //5
+        v( 1.0, -1.0,  1.0, position, scale), //6
+        v(-1.0, -1.0,  1.0, position, scale), //7
+        // top
+        v(-1.0,  1.0,  1.0, position, scale), //8
+        v(-1.0, -1.0,  1.0, position, scale), //9
+        v( 1.0, -1.0,  1.0, position, scale), //10
+        v( 1.0,  1.0,  1.0, position, scale), //11
+        // bottom
+        v(-1.0,  1.0, -1.0, position, scale), //12
+        v( 1.0,  1.0, -1.0, position, scale), //13
+        v( 1.0, -1.0, -1.0, position, scale), //14
+        v(-1.0, -1.0, -1.0, position, scale), //15
+        // right
+        v( 1.0,  1.0, -1.0, position, scale), //16
+        v( 1.0,  1.0,  1.0, position, scale), //17
+        v( 1.0, -1.0,  1.0, position, scale), //18
+        v( 1.0, -1.0, -1.0, position, scale), //19
+        // left
+        v(-1.0,  1.0, -1.0, position, scale), //20
+        v(-1.0, -1.0, -1.0, position, scale), //21
+        v(-1.0, -1.0,  1.0, position, scale), //22
+        v(-1.0,  1.0,  1.0, position, scale), //23
+    ];
+
+        #[rustfmt::skip]
+    let vertex_normals = vec![
+        // back
+        n( 0.0,  1.0,  0.0),
+        n( 0.0,  1.0,  0.0),
+        n( 0.0,  1.0,  0.0),
+        n( 0.0,  1.0,  0.0),
+        // front
+        n( 0.0, -1.0,  0.0),
+        n( 0.0, -1.0,  0.0),
+        n( 0.0, -1.0,  0.0),
+        n( 0.0, -1.0,  0.0),
+        // top
+        n( 0.0,  0.0,  1.0),
+        n( 0.0,  0.0,  1.0),
+        n( 0.0,  0.0,  1.0),
+        n( 0.0,  0.0,  1.0),
+        // bottom
+        n( 0.0,  0.0, -1.0),
+        n( 0.0,  0.0, -1.0),
+        n( 0.0,  0.0, -1.0),
+        n( 0.0,  0.0, -1.0),
+        // right
+        n( 1.0,  0.0,  0.0),
+        n( 1.0,  0.0,  0.0),
+        n( 1.0,  0.0,  0.0),
+        n( 1.0,  0.0,  0.0),
+        // left
+        n(-1.0,  0.0,  0.0),
+        n(-1.0,  0.0,  0.0),
+        n(-1.0,  0.0,  0.0),
+        n(-1.0,  0.0,  0.0),
+    ];
+
+        #[rustfmt::skip]
+    let faces = vec![
+        // back
+        TriangleFace::new(0, 1, 2),
+        TriangleFace::new(2, 3, 0),
+        // front
+        TriangleFace::new(4, 5, 6),
+        TriangleFace::new(6, 7, 4),
+        // top
+        TriangleFace::new(8, 9, 10),
+        TriangleFace::new(10, 11, 8),
+        // bottom
+        TriangleFace::new(12, 13, 14),
+        TriangleFace::new(14, 15, 12),
+        // right
+        TriangleFace::new(16, 17, 18),
+        TriangleFace::new(18, 19, 16),
+        // left
+        TriangleFace::new(20, 21, 22),
+        TriangleFace::new(22, 23, 20),
+    ];
+
+        Geometry::from_triangle_faces_with_vertices_and_normals(
+            faces,
+            vertex_positions,
+            vertex_normals,
+        )
+    }
+
+    pub fn cube_smooth_var_len(position: [f32; 3], scale: f32) -> Geometry {
```

Comment:
Same as above. Why duplicate this?

Reply:
I'll rename this one, as the normals are calculated differently to match the welding algorithm.